students/kate/session04/trigrams.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+"""
+
+trigrams with sherlock writing
+
+"""
+# import random module
+from random import randint
+
+# function to make pairs and followers in trigram dictionary
+def gen_trigrams(words):
+    # make an empty dicitonary for the trigrams
+    trigrams_dict = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i+2])
+        follower = words[i+2]
+        print(pair, follower)
+        if pair in trigrams_dict:
+            trigrams_dict[pair].append(follower)
+        trigrams_dict[pair] = [follower]
+    return(trigrams_dict)
+
+def build_new_text(word_dict):
+    # create index to hold a random integer
+    rand_index = randint(0, len(word_dict.keys()))
+    # create list to hold words
+    new_list = []
+    # set counter to 0
+    index = 0
+
+    # build sentences using index counter. Use for loop to add keys and values
+    for keys,values in word_dict.items():
+        if rand_index == index:
```

Comment:
this is pretty klunky -- you are looping through the entire dict until you get to the random index.

better to get it directly:

list(word_dict.keys())[rand_index]

or other options -- see:

https://github.com/UWPCE-PythonCert/IntroPython-2017/blob/master/solutions/Session04/arbitrary_key.py

(arbitrary_key.py in the solutions dir of the class repo)